File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/actions/fight/GameActionFightInvisibilityMessage.scala





















// Generated on 05/03/2013 10:34:33
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait GameActionFightInvisibilityMessage extends AbstractGameActionMessage {
  override val messageId = 5821

  protected trait Message extends AbstractGameActionMessage.Message with Serializable {
    override val messageId = GameActionFightInvisibilityMessage.messageId

    val targetId: Int
    val state: Byte

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeInt(targetId)
      buf.writeByte(state)
    }

  }

  protected def create(actionId: Short, sourceId: Int, targetId: Int, state: Byte): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val actionId = buf.readShort()
    val sourceId = buf.readInt()
    val targetId = buf.readInt()
    val state = buf.readByte()

    create(actionId, sourceId, targetId, state)
  }

}

object GameActionFightInvisibilityMessage extends GameActionFightInvisibilityMessage {
  case class MessageImpl(actionId: Short, sourceId: Int, targetId: Int, state: Byte) extends Message
  type Target = MessageImpl

  override def create(actionId: Short, sourceId: Int, targetId: Int, state: Byte) = MessageImpl(actionId, sourceId, targetId, state)

  def apply(actionId: Short, sourceId: Int, targetId: Int, state: Byte) = create(actionId, sourceId, targetId, state)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/GameActionFightInvisibilityMessage.java


// Generated on 05/08/2013 19:37:39
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GameActionFightInvisibilityMessage extends AbstractGameActionMessage {
    public static final int MESSAGE_ID = 5821;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int targetId;
    public byte state;
    
    public GameActionFightInvisibilityMessage() { }
    
    public GameActionFightInvisibilityMessage(short actionId, int sourceId, int targetId, byte state) {
        super(actionId, sourceId);
        this.targetId = targetId;
        this.state = state;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeInt(targetId);
        buf.writeByte(state);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        targetId = buf.readInt();
        state = buf.readByte();
    }
    
}
